bbconf/config_parser/bedbaseconfig.py

```diff
@@ -0,0 +1,338 @@
+        """
+
+        try:
+            return text2bednn.Text2BEDSearchInterface(
+                nl2vec_model=FlagEmbedding(model_name=self._config.path.text2vec),
+                vec2vec_model=self._config.path.vec2vec,
+                search_backend=self.qdrant_engine,
+            )
+        except Exception as e:
+            _LOGGER.error("Error in creating Text2BEDSearchInterface object: " + str(e))
+            warnings.warn(
+                "Error in creating Text2BEDSearchInterface object: " + str(e),
+                UserWarning,
+            )
+            return None
+
+    @staticmethod
+    def _init_pephubclient() -> Union[PEPHubClient, None]:
+        """
+        Create Pephub client object using credentials provided in config file
+
+        :return: PephubClient
+        """
+        try:
+            return PEPHubClient()
+        except Exception as e:
+            _LOGGER.error(f"Error in creating PephubClient object: {e}")
+            warnings.warn(f"Error in creating PephubClient object: {e}", UserWarning)
+            return None
+
+    def _init_boto3_client(
+        self,
+    ) -> boto3.client:
+        """
+        Create Pephub client object using credentials provided in config file
+
+        :return: PephubClient
+        """
+        try:
+            return boto3.client(
+                "s3",
+                endpoint_url=self._config.s3.endpoint_url,
+                aws_access_key_id=self._config.s3.aws_access_key_id,
+                aws_secret_access_key=self._config.s3.aws_secret_access_key,
+            )
+        except Exception as e:
+            _LOGGER.error(f"Error in creating boto3 client object: {e}")
+            warnings.warn(f"Error in creating boto3 client object: {e}", UserWarning)
+            return None
+
+    def _init_r2v_object(self) -> Region2VecExModel:
+        """
+        Create Region2VecExModel object using credentials provided in config file
+        """
+        return Region2VecExModel(self.config.path.region2vec)
+
+    def upload_s3(self, file_path: str, s3_path: Union[Path, str]) -> None:
+        """
+        Upload file to s3.
+
+        :param file_path: local path to the file
+        :param s3_path: path to the file in s3 with file name
+        :return: None
+        """
+        if not self._boto3_client:
+            _LOGGER.warning(
+                "Could not upload file to s3. Connection to s3 not established. Skipping.."
+            )
+            raise BedbaseS3ConnectionError(
+                "Could not upload file to s3. Connection error."
+            )
+        if not os.path.exists(file_path):
+            raise BedBaseConfError(f"File {file_path} does not exist.")
+        return self._boto3_client.upload_file(file_path, self.config.s3.bucket, s3_path)
+
+    def upload_files_s3(
+        self,
+        identifier: str,
+        files: Union[BedFiles, BedPlots, BedSetPlots],
+        base_path: str,
+        type: str = "files",
```

Comment:
Should this be an enum? I like typing these with `Listeral["files", "plots", ...]` because you get nice intellisense